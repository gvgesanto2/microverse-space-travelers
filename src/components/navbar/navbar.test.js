import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import Navbar from './navbar.component';

let component;

describe('Navbar component', () => {
  beforeEach(() => {
    component = render(<Navbar />, { wrapper: BrowserRouter });
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should navigate to the homepage when clicking the navbar\'s logo', () => {
    const logoLink = screen.getByRole('link', { name: /space travelers/i });

    userEvent.click(logoLink);

    expect(window.location.pathname).toBe('/');
  });

  it('should navigate to the Rockets Page when clicking the rockets navigation link', () => {
    const rocketsNavLink = screen.getByRole('link', { name: 'rockets' });

    userEvent.click(rocketsNavLink);

    expect(window.location.pathname).toBe('/');
  });

  it('should navigate to the Missios Page when clicking the Missions navigation link', () => {
    const missionsNavLink = screen.getByRole('link', { name: 'missions' });

    userEvent.click(missionsNavLink);

    expect(window.location.pathname).toBe('/missions');
  });

  it('should navigate to the MyProfile Page when clicking the my profile navigation link', () => {
    const myProfileNavLink = screen.getByRole('link', { name: 'my profile' });

    userEvent.click(myProfileNavLink);

    expect(window.location.pathname).toBe('/my-profile');
  });
});
