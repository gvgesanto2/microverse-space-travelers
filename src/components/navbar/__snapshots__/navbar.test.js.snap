// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="navbar"
      >
        <a
          aria-current="page"
          class="navbar__logo active"
          href="/"
        >
          <img
            alt="Space Travelers Logo"
            class="navbar__logo-img"
            src="logo.png"
          />
          <p
            class="navbar__logo-text"
          >
            Space Travelers' Hub
          </p>
        </a>
        <nav
          class="navbar__nav"
        >
          <ul
            class="navbar__nav-list"
          >
            <li>
              <a
                aria-current="page"
                class="navbar__nav-link navbar__nav-link--active"
                href="/"
              >
                rockets
              </a>
            </li>
            <li>
              <a
                class="navbar__nav-link"
                href="/missions"
              >
                missions
              </a>
            </li>
            <li>
              <a
                class="navbar__nav-link"
                href="/my-profile"
              >
                my profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <main
        class="main"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="navbar"
    >
      <a
        aria-current="page"
        class="navbar__logo active"
        href="/"
      >
        <img
          alt="Space Travelers Logo"
          class="navbar__logo-img"
          src="logo.png"
        />
        <p
          class="navbar__logo-text"
        >
          Space Travelers' Hub
        </p>
      </a>
      <nav
        class="navbar__nav"
      >
        <ul
          class="navbar__nav-list"
        >
          <li>
            <a
              aria-current="page"
              class="navbar__nav-link navbar__nav-link--active"
              href="/"
            >
              rockets
            </a>
          </li>
          <li>
            <a
              class="navbar__nav-link"
              href="/missions"
            >
              missions
            </a>
          </li>
          <li>
            <a
              class="navbar__nav-link"
              href="/my-profile"
            >
              my profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="main"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
